src/Operation/CreateCollection.php

```diff
@@ -214,6 +232,22 @@ public function __construct($databaseName, $collectionName, array $options = [])
             trigger_error('The "autoIndexId" option is deprecated and will be removed in a future release', E_USER_DEPRECATED);
         }
 
+        if (isset($options['pipeline'])) {
+            $expectedIndex = 0;
+
+            foreach ($options['pipeline'] as $i => $operation) {
+                if ($i !== $expectedIndex) {
+                    throw new InvalidArgumentException(sprintf('$pipeline is not a list (unexpected index: "%s")', $i));
+                }
+
+                if (! is_array($operation) && ! is_object($operation)) {
+                    throw InvalidArgumentException::invalidType(sprintf('$pipeline[%d]', $i), $operation, 'array or object');
+                }
+
+                $expectedIndex += 1;
+            }
+        }
```

Comment:
This was copied from Aggregate.php, but the exception messages were not updated to reflect that `pipeline` is an option and not a constructor parameter. Please apply the suggestions above.

I also created [PHPLIB-881](https://jira.mongodb.org/browse/PHPLIB-881) to refactor this to share common code with other instances of pipeline validation.

Reply:
Code updated in https://github.com/mongodb/mongo-php-library/pull/1078